app/main.py

```diff
@@ -2,5 +2,24 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result_dict = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+        for arg in args:
+            if (
```

Comment:
this if is redundant

Reply:
this if checks mutable types, then just return function without wrappin